pyriemann/spatialfilters.py

```diff
@@ -549,17 +736,15 @@ class AJDC(BaseEstimator, TransformerMixin):
     """
 
     def __init__(self, window=128, overlap=0.5, fmin=None, fmax=None, fs=None,
-                 expl_var=0.999, verbose=True):
+                 dim_red={'max_cond': 100}, verbose=True):
```

Comment:
I don't like this. To me you should be using a scikit-learn pipeline and not do any preprocessing in the init. It also makes the API more complex with dict which should not be a default parameter as it's a mutable object.

Reply:
What do you suggest? Defining `None` as default value for `dim_red`?

Reply:
I would not offer any dim red option in AJDC and tell users to use a pipeline with one step that does the dim red

Reply:
I don't see how to use a `sklearn` pipeline inside `AJDC`.
Because dimension reduction is an inner step of `AJDC.fit()`, applied between cospectra estimation and ajd_pham.

Can you point an example using the suggested architecture?